[Charlesl12] iP

Charlesl12 opened the pull request:
# Clovis

> "Your mind is for having ideas, not holding them." - David Allen [source](https://dansilvestre.com/productivity-quotes)

Clovis helps you to remember everything you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you have to do is,

1. download it from [here](https://github.com/Charlesl12/ip).
2. add your tasks
3. let it manage your tasks for you 👍 

And it is **FREE**!

Features:
- [X] Managing tasks
- [X] Managing deadlines
- [X] Managing events

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java  
public class Main {
    public static void main(String[] args) {
        Application.lauch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Charlesl12 committed:
rename main class, add greet, add exit

Charlesl12 committed:
add echo

Charlesl12 committed:
add list

Charlesl12 committed:
add task class, mark as done

Charlesl12 committed:
add todo.java, deadline.java, event.java

Charlesl12 committed:
add automated-test-ui-testing

Charlesl12 committed:
add clovisexception.java to handle errors

Charlesl12 committed:
add delete function

Charlesl12 committed:
add enums

Charlesl12 committed:
update testcases

Charlesl12 committed:
Add Storage.java

Charlesl12 committed:
Modify path for the use of Storage.java

Charlesl12 committed:
Merge branch 'branch-Level-7'

Charlesl12 committed:
Modify deadline.java and event.java to understand dates and times

Charlesl12 committed:
Merge branch 'branch-Level-8'

Bryce-3D commented on src/main/java/Clovis.java:
```diff
@@ -0,0 +1,109 @@
+        ArrayList<Task> list = Storage.loadTasks();
+        System.out.println("Hello! I'm Clovis.\nWhat can I do for you?");
+
+        enum Command {
```
Nice use of enums for the commands

Bryce-3D commented on src/main/java/Clovis.java:
```diff
@@ -0,0 +1,109 @@
+        while(true) {
+            try {
+                String input = br.readLine();
+                String[] split_input = input.split(" ", 2);
```
Hi! I think this variable should be renamed as something like `splitInput` since the codestyle says that in general, variables should be written in `camelCase`. It also says that collections of stuff like lists should be in plural form, but given the variable here, I'm not really sure if `splitInputs` is exactly accurate either.

Bryce-3D commented on src/main/java/Clovis.java:
```diff
@@ -0,0 +1,109 @@
+                String[] split_input = input.split(" ", 2);
+                Command command = Command.fromString(split_input[0]);
+
+                switch(command) {
```
Hi! The codestyle guide for CS2103T mentions that the switch case block should not have an extra indent for the case clauses, so it should look something like this
```java
switch (cmd) {
case LIST:
    // Something
    break;
case BYE:
    // Something
    // ...
```

Bryce-3D commented on src/main/java/Clovis.java:
```diff
@@ -0,0 +1,109 @@
+                        System.out.println("Bye. Hope to see you again soon!");
+                        return;
+                    case MARK:
+                        int mark_index = Integer.parseInt(split_input[1]);
```
Hi! I think this is another variable that does not follow the `camelCase` naming convention, and I see quite a few more below. I think it might be a good idea to ctrl+F for `_`s in your code to find and rename variables from `snake_case` to `camelCase`. I also made this mistake at first and had to spend a bit of time fixing it a week or two ago.

Bryce-3D commented on src/main/java/Clovis.java:
```diff
@@ -0,0 +1,109 @@
+                        } else if (taskType.equalsIgnoreCase("event")) {
+                            String[] splitDescription = description.split("/from | /to ");
+                            if (description.isEmpty() || splitDescription.length < 3) {
+                                throw new ClovisException("Unacceptable, a description, /from and /to date are required for an event");
```
Hi! I think this line, when including the indent, exceeds 120 characters. In the CS2103T style guide, it mentions 110 chars as a soft limit and 120 chars as a hard limit for the line length, so maybe a line break can be added somewhere.

Bryce-3D commented on src/main/java/Clovis.java:
```diff
@@ -0,0 +1,109 @@
+                        } else if (taskType.equalsIgnoreCase("deadline")) {
+                            String[] splitDescription = description.split("/by ", 2);
+                            if (description.isEmpty() || splitDescription.length < 2) {
+                                throw new ClovisException("Unacceptable, a description and /by date are required for a deadline");
```
Hi! I think this line, like the one below, likely exceeds the 120 chars hard limit for line length

Bryce-3D commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + "(from: " + start.format(OUTPUT_FORMAT) + " to: " + end.format(OUTPUT_FORMAT) + ")";
```
Hi! I think that this line also exceeds 120 chars, so a line break would be needed here too.

Bryce-3D reviewed (commented):
Hi! Another general comment is that there is currently no javadocs available. As listed under the [`A-JavaDocs` task from week 3](https://nus-cs2103-ay2425s2.github.io/website/schedule/week3/project.html#5-add-increments-as-parallel-branches-a-javadoc-a-codingstandard-level-9)
> Minimal: Add header comments to at least half of the non-private classes/methods.

I hope this review helps!

Charlesl12 committed:
Add Ui.java, TaskList.java, Storage.java, Command.java and its subclasses

TasmiaH0508 commented on src/main/java/Clovis.java:
```diff
@@ -0,0 +1,109 @@
+import java.time.LocalDateTime;
+
+public class Clovis {
+    public static void main(String[] args) throws IOException {
```
Perhaps it would be better to group chunks of code and turn them into methods so that readability of the code is improved in general.

TasmiaH0508 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Event extends Task {
```
I think it would be good to have JavaDoc comments for as many methods and classes as possible.

TasmiaH0508 reviewed (commented):
Generally, there are very few errors in following the java coding standard. 

Code quality, in the naming aspect, is good as descriptive names were given to variables and functions. 

However, there is a lack of JavaDoc comments.

Charlesl12 committed:
Refactor Clovis.java

Charlesl12 committed:
Add packages

Charlesl12 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Charlesl12 committed:
Use gradle

Charlesl12 committed:
Add JUnit tests for ToDo.java and TaskList.java

Charlesl12 committed:
Package as app as a JAR file

Charlesl12 committed:
Refactor code

Charlesl12 committed:
Add JavaDocs for Clovis.java, Parser.java, Ui.java, Storage.java, ClovisException.java

Charlesl12 committed:
Add JavaDocs for TaskList.java, Task.java and its subclasses

Charlesl12 committed:
Add JavaDocs to all the Command classes

Charlesl12 committed:
Add findCommand.java

Charlesl12 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/clovis/Parser.java
#	src/main/java/clovis/Storage.java
#	src/main/java/clovis/command/AddCommand.java
#	src/main/java/clovis/command/Command.java
#	src/main/java/clovis/command/UnmarkCommand.java

Charlesl12 committed:
Merge branch 'branch-Level-9'

Charlesl12 committed:
Use CheckStyle and made checkstyle changes to all classes

Charlesl12 committed:
Add JavaDocs for FindCommand.java

Charlesl12 committed:
Add GUI for Clovis

Charlesl12 committed:
Add GUI for Clovis

(cherry picked from commit 5b6847dc26ed4bffa14f917d8a9403b07122cd89)

Charlesl12 committed:
Refactor code and JavaDocs

(cherry picked from commit ef011b0e722e73b42db452e30671164e12aac695)

Charlesl12 committed:
Refactor code and JavaDocs

Charlesl12 committed:
Merge branch 'branch-level-10'

Charlesl12 committed:
Add assertions for Parser.java, TaskList.java, Clovis.java and all Command subclasses

Charlesl12 committed:
Refactor code

Charlesl12 committed:
Add use of streams to Storage.java

Charlesl12 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/clovis/command/DeleteCommand.java
#	src/main/java/clovis/command/MarkCommand.java
#	src/main/java/clovis/command/UnmarkCommand.java

Charlesl12 committed:
Merge branch 'branch-A-Streams'

Charlesl12 committed:
Remove repeated assertions

Charlesl12 committed:
Add JUnit test for FindCommand.java

Charlesl12 committed:
TaskList.java: fix findTask method to be case-insensitive

The method is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Converting both the retrieval of task of the list and the input from the
users to lower letter case, checking and comparing of these strings
are more consistent and made case-insensitive.

Let's,
* update the search algorithm to use case-insensitive matching
* add a junit test to the new format

Charlesl12 committed:
Add use of streams of Storage.java

Charlesl12 committed:
Merge pull request #3 from Charlesl12/branch-A-Streams

Add use of streams of Storage.java

Charlesl12 committed:
Merge branch 'branch-A-Streams'

Charlesl12 committed:
Merge branch 'master' of https://github.com/Charlesl12/ip

Charlesl12 committed:
Add use of streams in Storage.java (reversed the commit wrongly)

Charlesl12 committed:
Improve application to deal with duplicate tasks.

Add tests to test the function above.

Charlesl12 committed:
Improve application to deal with schedule anomalies

Charlesl12 committed:
Adhere to checkstyle

Charlesl12 committed:
Add Ui.png

Fix Deadline and Event classes check for past dates as inputs

Charlesl12 committed:
Add user guide, reupload Ui.png

Charlesl12 committed:
Update user guide

Charlesl12 committed:
Bug fix: non-integer index and empty index cause application to crash
